monai/transforms/post/array.py

```diff
@@ -128,6 +132,9 @@ class AsDiscrete(Transform):
             Defaults to ``None``.
         rounding: if not None, round the data according to the specified option,
             available options: ["torchrounding"].
+        kwargs: additional parameters to `torch.argmax`, `monai.networks.one_hot`.
```

Comment:
Maybe it's more clear to use `dtype` and `kwargs` as 2 args? Then:
```py
torch.argmax(..., **kwargs)
onehot(..., dtype=self.dtype)
```
What do you think?

Thanks.

Reply:
post_pred = Compose([EnsureType(), Activations(softmax=True)])
post_label = Compose([EnsureType(), AsDiscrete(to_onehot=2)])

then i am evaluating here

 acc_value = torch.eq(y_pred.argmax(dim=1), y)
        acc_metric = acc_value.sum().item() / len(acc_value)
        y_onehot = [post_label(i) for i in decollate_batch(y)]
        y_pred_act = [post_pred(i) for i in decollate_batch(y_pred)]
        auc_metric(y_pred_act, y_onehot)
        auc_result = auc_metric.aggregate()
        auc_metric.reset()
        del y_pred_act, y_onehot

it shows this error
applying transform <monai.transforms.post.array.AsDiscrete object at 0x7f48e0027a90>

Reply:
I didn't understand what you are trying to say.

Reply:
This is also given in [3d classification](https://github.com/Project-MONAI/tutorials/blob/master/3d_classification/torch/densenet_training_dict.py)

Reply:
AssertionError: One of self.n_classes or n_classes must be an integer
This also shows this

Reply:
The above exception was the direct cause of the following exception:
raise RuntimeError(f"applying transform {transform}") from e